packages/devtools_app/lib/src/utils.dart

```diff
@@ -1069,3 +1069,225 @@ String prettyTimestamp(
   );
   return DateFormat.Hms().format(timestampDT); // HH:mm:ss
 }
+
+/// A [ChangeNotifier] that holds a list of data.
+///
+/// This class also exposes methods to interact with the data. By default,
+/// listeners are notified whenever the data is modified, but notifying can be
+/// optionally disabled.
+class ListValueNotifier<T> extends ChangeNotifier
+    implements ValueListenable<List<T>> {
+  /// Creates a [ListValueNotifier] that wraps this value [_rawList].
+  ListValueNotifier(List<T> rawList) : _rawList = List<T>.from(rawList) {
+    _currentList = ImmutableList(_rawList);
+  }
+
+  final List<T> _rawList;
+
+  ImmutableList<T> _currentList;
+
+  @override
+  List<T> get value => _currentList;
+  set value(List<T> newValue) {
+    throw Exception(
+      '[value] cannot be set explicitly. Use the methods on ListValueNotifier'
+      ' to interact with [value].',
+    );
+  }
+
+  void add(T item, {bool notify = true}) {
+    _rawList.add(item);
+    _currentList = ImmutableList(_rawList);
+    if (notify) {
+      notifyListeners();
+    }
+  }
+
+  void addAll(List<T> itemList, {bool notify = true}) {
+    _rawList.addAll(itemList);
+    _currentList = ImmutableList(_rawList);
+    if (notify) {
+      notifyListeners();
+    }
+  }
+
+  bool remove(T element, {bool notify = true}) {
+    final removed = _rawList.remove(element);
+    _currentList = ImmutableList(_rawList);
+    if (removed && notify) {
+      notifyListeners();
+    }
+    return removed;
+  }
+
+  T removeAt(int index, {bool notify = true}) {
+    final element = _rawList.removeAt(index);
+    _currentList = ImmutableList(_rawList);
+    if (notify) {
+      notifyListeners();
+    }
+    return element;
+  }
+
+  void removeWhere(bool test(T element), {bool notify = true}) {
+    _rawList.removeWhere(test);
+    _currentList = ImmutableList(_rawList);
+    if (notify) {
+      notifyListeners();
+    }
+  }
+
+  void insert(int index, T element, {bool notify = true}) {
+    _rawList.insert(index, element);
+    _currentList = ImmutableList(_rawList);
+    if (notify) {
+      notifyListeners();
+    }
+  }
+
+  void insertAll(int index, Iterable<T> iterable, {bool notify = true}) {
+    _rawList.insertAll(index, iterable);
+    _currentList = ImmutableList(_rawList);
+    if (notify) {
+      notifyListeners();
+    }
+  }
+
+  void clear({bool notify = true}) {
```

Comment:
this needs to set
_rawList = [];
instead of clearing. Otherwise you are breaking existing immutable values.